scale/storage/brokers/broker.py

```diff
@@ -137,23 +144,183 @@ def upload_files(self, upload_dir, work_dir, files_to_upload):
         :param files_to_upload: List of tuples (source path relative to upload directory, workspace path for storing the
             file)
         :type files_to_upload: list of (str, str)
-        '''
+        """
 
         pass
 
     def _validate_str_config_field(self, name, config):
-        '''Validates that the given field name has a non-empty string value in given configuration. There is no return
+        """Validates that the given field name has a non-empty string value in given configuration. There is no return
         value; an invalid configuration should just raise an exception.
 
         :param name: The name of the field
         :type name: str
         :param config: The configuration as a dictionary
         :type config: dict
-        '''
+        """
         if not name in config:
             msg = u'%s is required for %s broker'
             raise Exception(msg % (name, self.broker_type))
 
         if not config[name] or not isinstance(config[name], basestring):
             msg = u'%s must have a non-empty string value for %s broker'
             raise Exception(msg % (name, self.broker_type))
+
+
+class Broker(object):
+    """Abstract class for a broker that can download and upload files for a given storage backend
+    """
+
+    __metaclass__ = ABCMeta
+
+    def __init__(self, broker_type):
+        """Constructor
+
+        :param broker_type: The type of this broker
+        :type broker_type: string
+        """
+
+        self._broker_type = broker_type
+        self._mount = None
+        self._uses_mount = False
+
+    @property
+    def broker_type(self):
+        """The type of this broker
+
+        :returns: The broker type
+        :rtype: string
+        """
+
+        return self._broker_type
+
+    @property
+    def mount(self):
+        """If this broker uses a mounted file system (uses_mount is True), this property returns the remote location
+        that should be mounted into the task container. If uses_mount is False, this property should be None.
+
+        :returns: The remote file system location to mount, possibly None
+        :rtype: string
+        """
+
+        return self._mount
+
+    @property
+    def uses_mount(self):
```

Comment:
Seems like uses_mount() could be deleted and just have mount() return a value or None only. Since the documentation requires that behavior anyway, it's one less thing to keep in sync.